### PR TITLE
4 hossein fix right panel scroll

### PR DESCRIPTION
**Pull request by hoseinmirian**
in progress of simplifying scroll..

@terazus @knirirr 
updates in this branch
- better header behavior
- refactoring some codes
- JumpToTop refactoring
- Refactored Scroll

### CODE REVIEW

**Comment by hoseinmirian** [comment]
in this branch I try to remove some of the complexities of the app such as overflow variable used to control the body overflow of the app for the sake of scroll. Now scroll is replaced with a window scroll causes reduction in complexity of testing the app. Also I have not yet shown Footer in this branch and that one will be fixed in 4.1 branch. Please note also you might see double Filter chips and not having tests fully passed which all will be fixed in 4.1 branch.

File: src/views/Records/Records.vue [hunk 1]
```diff
@@ -58,6 +58,7 @@
           />
         </v-col>
       </v-row>
+      <Footer class="mb-2" />
```
Comment (terazus) [inline src/views/Records/Records.vue]: Im unsure why this is here and not in App.vue like discussed in a previous PR.

File: tests/unit/components/Records/Search/Header/FilterChips.spec.js [hunk 2]
```diff
@@ -39,12 +39,8 @@ describe("FilterChips.vue", () => {
         wrapper.vm.$router.push = push;
         await wrapper.vm.removeParam("publications", "pub2");
         expect($router.push).toHaveBeenCalledTimes(1);
-        let query = wrapper.vm.buildNewQuery("grants", "ABCDEF");
-        expect(query).toStrictEqual({page: 1, publications: 'pub1,pub2,pub3'});
-        query = wrapper.vm.buildNewQuery("publications", "pub4");
-        expect(query).toStrictEqual({page: 1, grants: 'ABCDEF', publications: 'pub1,pub2,pub3'});
-        await wrapper.vm.removeParam("publications", "pub3");
-        expect($router.push).toHaveBeenCalledTimes(1);
```
Comment (terazus) [inline tests/unit/components/Records/Search/Header/FilterChips.spec.js]: Why did you remove these tests?

**Review by terazus — CHANGES_REQUESTED** [review]

File: tests/unit/components/Records/Search/Header/FilterChips.spec.js (hunk 2 above)
Comment (hoseinmirian) [reply tests/unit/components/Records/Search/Header/FilterChips.spec.js]: by confusion! unfortunately they were taken out but are put back in 4.1

File: src/views/Records/Records.vue (hunk 1 above)
Comment (hoseinmirian) [reply src/views/Records/Records.vue]: it is fixed in update branch 4.1

**Review by knirirr — CHANGES_REQUESTED** [review]
I'm seeing problems with tests on this, unfortunately. The output is attached.

[test_error.txt](https://github.com/FAIRsharing/fairsharing.github.io/files/5652367/test_error.txt)

**Review by terazus — CHANGES_REQUESTED** [review]
Please fix your tests:
https://github.com/FAIRsharing/fairsharing.github.io/runs/1499789075

**Review by knirirr — CHANGES_REQUESTED** [review]
Clicking the scroll-to-top button on a record page produces the following in the console (and also doesn't scroll):

```
[Vue warn]: Error in v-on handler: "TypeError: can't access property "scrollTo", target is null"

found in

---> <VBtn>
       <JumpToTop> at src/components/Navigation/jumpToTop.vue
         <VMain>
           <Records> at src/views/Records/Records.vue
             <VApp>
               <App2> at src/App.vue
                 <Root> vue.runtime.esm.js:619
TypeError: can't access property "scrollTo", target is null
    gotoTop navigationUtils.js:2
    scrollToTop jumpToTop.vue:28
    click jumpToTop.vue:22
    VueJS 4
    click VBtn.ts:158
    VueJS 3
```

**Comment by hoseinmirian** [comment]
> 
> 
> Clicking the scroll-to-top button on a record page produces the following in the console (and also doesn't scroll):
> 
> ```
> [Vue warn]: Error in v-on handler: "TypeError: can't access property "scrollTo", target is null"
> 
> found in
> 
> ---> <VBtn>
>        <JumpToTop> at src/components/Navigation/jumpToTop.vue
>          <VMain>
>            <Records> at src/views/Records/Records.vue
>              <VApp>
>                <App2> at src/App.vue
>                  <Root> vue.runtime.esm.js:619
> TypeError: can't access property "scrollTo", target is null
>     gotoTop navigationUtils.js:2
>     scrollToTop jumpToTop.vue:28
>     click jumpToTop.vue:22
>     VueJS 4
>     click VBtn.ts:158
>     VueJS 3
> ```

@knirirr .It is covered in the next branch No 5 which is now in the pr lists. Please only focus on the title of the branch, each branch focuses on one specific issue to keep the branch smaller, tnx,

**Comment by knirirr** [comment]
> @knirirr .It is covered in the next branch No 5 which is now in the pr lists. Please only focus on the title of the branch, each branch focus on one specific issue to keep it shorter. tnx,

If there's a bug in the code then there a PR should not be created; much as large PRs with multiple features in are hard to review, so are multiple PRs for one small feature.

**Review by knirirr — CHANGES_REQUESTED** [review]
Well, it seems to work in a browser. I do see this message in the Firefox console, though:

```quote
This site appears to use a scroll-linked positioning effect. This may not work well with asynchronous panning; see https://developer.mozilla.org/docs/Mozilla/Performance/ScrollLinkedEffects for further details and to join the discussion on related tools and features!
```
Do you also see this when running the client locally?

```

/home/milo/Work/Web/fairsharing.github.io/src/components/Navigation/Header.vue
  98:3  warning  The "data" property should be above the "computed" property on line 94
```

I would also advise merging the latest dev into your branch as that would remove some local search coverage problems.

**Comment by hoseinmirian** [comment]
> 
> 
> Well, it seems to work in a browser. I do see this message in the Firefox console, though:
> 
> ```
> This site appears to use a scroll-linked positioning effect. This may not work well with asynchronous panning; see https://developer.mozilla.org/docs/Mozilla/Performance/ScrollLinkedEffects for further details and to join the discussion on related tools and features!
> ```
> 
> Do you also see this when running the client locally?
> 
> ```
> 
> /home/milo/Work/Web/fairsharing.github.io/src/components/Navigation/Header.vue
>   98:3  warning  The "data" property should be above the "computed" property on line 94
> ```
> 
> I would also advise merging the latest dev into your branch as that would remove some local search coverage problems.

about the first point its fine its just a warnning due to we are working with several scrolls. About the second point I did not see any issue in the test locally but in the webStorm was somehow I was warnned by it. thanks for spotting the issue.
